leshan-server-core/src/main/java/org/eclipse/leshan/server/observation/ObservationListener.java

```diff
@@ -50,7 +50,7 @@
      * @param response the lwm2m response received (successful or error response)
      * 
      */
-    void onResponse(SingleObservation observation, Registration registration, ObserveResponse response);
+    void onResponse(Observation observation, Registration registration, AbstractLwM2mResponse response);
```

Comment:
You finally go in a different way than what we talk about in https://github.com/eclipse/leshan/issues/1034#issuecomment-876411280

Is it intended ? any reason about this ?

Reply:
It's rather my misunderstanding your intentions. By the comment https://github.com/eclipse/leshan/issues/1034#issuecomment-877268619 I understand that we should avoid methods multiplying and we can accept casting.

Do you think I should introduce two versions of onResponse() ?

Reply:
> Do you think I should introduce two versions of onResponse() ?

I feel this would be a better API. WDYT ?